### PR TITLE
[Woo POS][Cash & Receipts] Delegate receipt logic out of aggregate model

### LINKED ISSUE #14609: [Woo POS] [Cash & Receipts] Encapsulate and move receipt-related logic outside the aggregate model
Follow-up from https://github.com/woocommerce/woocommerce-ios/pull/14587#discussion_r1869507087 and p1733995434442479-slack-C070SJRA8DP

Thoughts:
- Receipt eligibility should be out of the aggregate model
- Since we're manipulating the order and relying on that order state, the core logic probably wants to go on the order controller, but not necessarily in the order service. The order controller could have both order service and receipts service, these should be simple and stateless.
- Views shouldn't have controllers
- We're currently limited by the API, which forces us to perform requests to send the receipt: the endpoint accepts an orderID and a siteID, but no string for the recipient email, as core uses the email that has already been added to the order, which in our case (POS) is none, so we need both APIs. Ideally we should need only one request by passing the email only.

### PR DESCRIPTION
**Pull request by iamgabrielma**
Closes: #14609 

## Description
This PR moves the receipt logic and the eligibility criteria outside of the aggregate model to its own service. Then the order controller delegates to each service (order and receipts) the tasks to update the order and send the receipts separately. For this we:

- Move receipt eligibility criteria to existing `ReceiptEligibilityUseCase`
- Introduce the `POSReceiptServiceProtocol` 
- Create a `POSReceiptService`

There are no functional changes to the underlying receipt logic, everything should work as always.

~~It seems the WooCommerce version needed for receipts is no longer 9.5 but 9.6, I made the changes in this PR as well but I might cherry-pick it separately. We're still debugging this bit here: p1734328456365229-slack-C070SJRA8DP~~ 

In order to receive emails in your test store, update the "From" address with your site's domain, under WooCommerce -> Settings -> Emails. Doc: https://woocommerce.com/document/configuring-woocommerce-settings/emails/

Screenshot 2024-12-16 at 14 35 14

### Case 1: Flag off.
- Run the app. Process a POS transaction. Observe the `Receipt` button does not render at the end of the flow.

Simulator Screenshot - iPad Air 11 - iOS 17 5 M2 - 2024-12-16 at 12 43 32

### Case 2: Flag on. Non-eligible store (WooCommerce <9.6)
- Switch `.sendReceiptsForPointOfSale` to `true`
- Run the app. Process a POS transaction. Observe the `Receipt` button  renders at the end of the flow.
- When tapping the `Receipt` button, you should see a toast that indicates this store does not support receipts yet.

Simulator Screenshot - iPad Air 11 - iOS 17 5 M2 - 2024-12-16 at 13 04 38

### Case 3. Flag on. Eligible store (WooCommerce 9.6+)
- Switch `.sendReceiptsForPointOfSale` to `true`
- Use a test site with WooCommerce 9.5+ (current available update to 9.5.0-rc.1)
- Process a POS transaction. Observe the `Receipt` button renders, tapping on it presents the receipts view. Enter a valid email address and tap on send. Confirm the email is received in your email.

simulator_screenshot_A148ABB3-104E-456C-8FC8-BC42D75AE622

---
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on all devices (phone/tablet) and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">View files have been modified, but no screenshot or video is included in the pull request. Consider adding some for clarity.</td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          1 Message
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">This PR is still a Draft: some checks will be skipped.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/WooCommerceTests/POS/Mocks/MockPOSOrderService.swift [hunk 1]
```diff
@@ -19,7 +19,9 @@ class MockPOSOrderService: POSOrderServiceProtocol {
         return order
     }
 
-    func sendReceipt(order: Yosemite.Order, recipientEmail: String) async throws { }
+    func updatePOSOrder(order: Order, recipientEmail: String) async throws {
+        // no-op
```
Comment (staskus) [inline WooCommerce/WooCommerceTests/POS/Mocks/MockPOSOrderService.swift]: We could consider updating PointOfSaleOrderController tests to confirm that both update and sendReceipt are called 👍

**Review by staskus — APPROVED** [review]
Thanks! Works well! 👍

File: WooCommerce/WooCommerceTests/POS/Mocks/MockPOSOrderService.swift (hunk 1 above)
Comment (iamgabrielma) [reply WooCommerce/WooCommerceTests/POS/Mocks/MockPOSOrderService.swift]: Good point, added: https://github.com/woocommerce/woocommerce-ios/pull/14663/commits/51cab4d2ee219a5a8c37fc4e7681d6736c113c97

**Pull request merged**